File: 0933.number_of_recent_calls/solution.java
class RecentCounter {
    private Queue<Integer> q;

    public RecentCounter() {
        q = new LinkedList<>();
    }

    public int ping(int t) {
        while (!q.isEmpty() && t - 3000 > q.peek()){
            q.poll();
        }
        q.add(t);
        return q.size();
    }
}

File: 0933.number_of_recent_calls/solution.py
class RecentCounter:
    def __init__(self):
        self.nums = []

    def ping(self, t: int) -> int:
        while self.nums and t - 3000 > self.nums[0]:
            self.nums.pop(0)
        self.nums.append(t)
        return len(self.nums)
